### ISSUE TITLE
changetype number to bool

### ISSUE THREAD

**Issue by LiaoPeng** [AssemblyScript/assemblyscript#342]
[The assemblyscript code](https://webassembly.studio/?f=jbbkyk6d76)

```
assert(true == changetype<boolean>(1));
assert(false == changetype<boolean>(0));
assert(true == changetype<boolean>(3));
assert(false == changetype<boolean>(2));
```
I expected:
assert(false == changetype<boolean>(2)); // failed

In fact, all asserts passed.

It seemed that `changetype` number to bool  just get the last bit.

**Comment by dcodeIO** [comment]
Note that `changetype` isn't meant for casting, but actually for changing between a reference and a value type, i.e. between `SomeObject` and `usize`.

For casting, there is `<bool>2`, or its portable version `bool(2)`. Also, conversion to `bool` is performed through `& 1` currently, which might be unexpected and might need to be changed to `!= 0` maybe.

**Comment by MaxGraey** [comment]
I think change this to `!= 0` is right apprach

**Comment by dcodeIO** [comment]
Does C/C++ handle `bool` specifically as `!= 0`?

**Comment by MaxGraey** [comment]
Currently `assert(true == <bool>(2));` is fail btw. Rust and C++ use `!= 0` approach.

```
0b10 & 1 = 0 // false
```

**Comment by dcodeIO** [comment]
Ok, so we should add a special case for bool I guess.

**Comment by dcodeIO** [comment]
Fixed in #343 for casts. Please keep in mind that `changetype` isn't meant for this.

**Comment by LiaoPeng** [comment]
cool
